maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/CheckoutMojoTest.java

```diff
@@ -81,6 +81,12 @@ public void testSkipCheckoutWithConnectionUrl()
 
         SvnScmTestUtils.initializeRepository( repository );
 
+        if ( !ScmTestCase.isSystemCmd( SvnScmTestUtils.SVN_COMMAND_LINE ) )
+        {
+            ScmTestCase.printSystemCmdUnavail( SvnScmTestUtils.SVN_COMMAND_LINE, getName() );
+            return;
+        }
+
```

Comment:
- I see this code fragment duplicated many times. Should we make this into a method that returns a boolean?
- If svn is missing then this test will pass. Now tests can pass, fail or be skipped. In many of the cases here I would have chosen to mark them as skipped instead of (as I read the code) passed.

Reply:
There is a problem, I have a local branch which uses your new code:
```
diff --git a/maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/BranchMojoTest.java b/maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/BranchMojoTest.java
index ef788478..33d27e7f 100644
--- a/maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/BranchMojoTest.java
+++ b/maven-scm-plugin/src/test/java/org/apache/maven/scm/plugin/BranchMojoTest.java
@@ -27,6 +27,9 @@

 import java.io.File;

+import static org.apache.maven.scm.provider.svn.SvnScmTestUtils.SVN_COMMAND_LINE;
+import static org.apache.maven.scm.provider.svn.SvnScmTestUtils.SVNADMIN_COMMAND_LINE;
+
 /**
  * @author <a href="mailto:evenisse@apache.org">Emmanuel Venisse</a>
  *
@@ -51,11 +54,7 @@ protected void setUp()

         FileUtils.forceDelete( repository );

-        if ( !ScmTestCase.isSystemCmd( SvnScmTestUtils.SVNADMIN_COMMAND_LINE ) )
-        {
-            ScmTestCase.printSystemCmdUnavail( SvnScmTestUtils.SVNADMIN_COMMAND_LINE, "setUp" );
-            return;
-        }
+        ScmTestCase.checkScmPresence( SVNADMIN_COMMAND_LINE );

         SvnScmTestUtils.initializeRepository( repository );
```

But all of these changes fail with `AssertionViolationException`. There is some old code in Plugin Testing which does not work with `Assume`. Therefore, I'd like to move on with this and analyze it in a subsequent ticket: https://issues.apache.org/jira/projects/SCM/issues/SCM-939

Reply:
Ok, then I fully agree with your current code.